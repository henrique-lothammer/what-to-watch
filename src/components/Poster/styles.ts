import styled from 'styled-components'

import { colors } from 'styles/variables'

interface IContainerProps {
  list: number
}

export const Container = styled.li`
  position: relative;
  width: 185px;
  height: 330px;
  display: flex;
  flex-direction: column;
  background: ${colors.bgLighter};
  ${(props: IContainerProps) =>
    props.list ? 'margin: 5px;' : 'margin-left: 15px;'}

  cursor: pointer;
  &:first-child {
    margin-left: 0;
  }
`

export const Picture = styled.div`
  width: 185px;
  height: 255px;
  position: relative;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  overflow: hidden;
`

export const Title = styled.article`
  color: ${colors.font};
  font-weight: bold;
  width: 100%;
  height: 75px;
  padding: 10px 5px 5px 5px;
  display: flex;
  align-items: center;
  justify-content: center;
  h3 {
    font-size: 1.5rem;
    text-align: center;
    overflow: hidden;
  }
`

export const Score = styled.div`
  background: ${colors.primary};
  color: ${colors.font};
  position: absolute;
  padding: 5px;
  border-radius: 100%;
  top: 5px;
  right: 5px;
  z-index: 2;
  width: 36px;
  height: 36px;
  text-align: center;
  text-align: center;
  font-size: 1.6rem;
  line-height: 2.6rem;
  box-shadow: 0 0 1em #000;
`
