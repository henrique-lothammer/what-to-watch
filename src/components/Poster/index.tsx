import React, { ReactElement } from 'react'
import { useHistory } from 'react-router-dom'

import Score from 'components/Score'

import noPicture from 'assets/no-picture.jpg'

import { IMovieList } from '../PostersSection/types'
import { Container, Title, Picture } from './styles'

export interface IProp {
  movie: IMovieList
  list?: boolean
}

const PosterSection = ({ movie, list = false }: IProp): ReactElement => {
  const history = useHistory()
  const { title, voteAverage } = movie
  return (
    <Container
      onClick={() => history.push(`/details/${movie.id}`)}
      list={list ? 1 : 0}
      title={movie.title}
    >
      <Score score={voteAverage} shadow />
      <Picture>
        <img
          src={
            movie.posterPath
              ? `${process.env.REACT_APP_TMDB_IMAGE_URL}/w185${movie.posterPath}`
              : noPicture
          }
          alt={movie.title}
        />
      </Picture>
      <Title>
        <h3>{title}</h3>
      </Title>
    </Container>
  )
}

export default PosterSection
