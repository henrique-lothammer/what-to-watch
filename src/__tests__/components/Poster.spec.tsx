import React from 'react'
import { render, cleanup, fireEvent } from '@testing-library/react'
import { createMemoryHistory } from 'history'
import { Router } from 'react-router-dom'
import Poster from 'components/Poster'
import 'jest-styled-components'

describe('Poster Component', () => {
  afterEach(cleanup)
  const movie = {
    id: 1,
    title: 'Movie test',
    posterPath: 'teste.path',
    voteAverage: 8,
  }

  it('should render', () => {
    const history = createMemoryHistory()
    const { container, getByText } = render(
      <Router history={history}>
        <Poster movie={movie} />
      </Router>
    )

    const element = getByText(/Movie test/i)
    expect(element).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })

  it('should navigate to the details page when clicked', () => {
    const history = createMemoryHistory()
    const { getByText } = render(
      <Router history={history}>
        <Poster movie={movie} />
      </Router>
    )
    fireEvent.click(getByText(/Movie test/i))
    expect(history.location.pathname).toEqual('/details/1')
  })
})
