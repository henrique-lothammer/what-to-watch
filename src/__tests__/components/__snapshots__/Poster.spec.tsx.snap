// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Poster Component should render 1`] = `
.c1 {
  background: #6e9a0b;
  color: #FFF;
  position: absolute;
  padding: 5px;
  border-radius: 100%;
  top: 10px;
  right: 5px;
  z-index: 2;
  width: 36px;
  height: 36px;
  text-align: center;
  text-align: center;
  font-size: 1.6rem;
  line-height: 2.6rem;
  box-shadow: 0 0 1em #000;
}

.c0 {
  position: relative;
  width: 185px;
  height: 330px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  background: #196774;
  margin-left: 15px;
  cursor: pointer;
}

.c0:first-child {
  margin-left: 0;
}

.c0:hover img {
  -webkit-transform: scale(1.05);
  -ms-transform: scale(1.05);
  transform: scale(1.05);
}

.c0:hover article {
  background: #6e9a0b;
}

.c2 {
  width: 185px;
  height: 255px;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  overflow: hidden;
}

.c2 img {
  -webkit-transition: 0.5s;
  transition: 0.5s;
  -webkit-transform: scale(1);
  -ms-transform: scale(1);
  transform: scale(1);
}

.c3 {
  color: #FFF;
  background: #196774;
  font-weight: bold;
  width: 100%;
  height: 75px;
  padding: 10px 5px 5px 5px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-transition: 0.5s;
  transition: 0.5s;
}

.c3 h3 {
  font-size: 1.5rem;
  text-align: center;
  overflow: hidden;
}

<li
  class="c0"
  list="0"
  title="Movie test"
>
  <div
    class="c1"
    data-testid="score"
  >
    8
  </div>
  <div
    class="c2"
  >
    <img
      alt="Movie test"
      src="https://image.tmdb.org/t/p/w185teste.path"
    />
  </div>
  <article
    class="c3"
  >
    <h3>
      Movie test
    </h3>
  </article>
</li>
`;
